dpx/src/dpx_dpxutil.rs

```diff
@@ -60,21 +60,20 @@ pub struct ht_iter {
    Copyright 2016-2018 the Tectonic Project
    Licensed under the MIT License.
 */
-#[no_mangle]
-pub unsafe extern "C" fn xtoi(mut c: i8) -> i32 {
-    if c as i32 >= '0' as i32 && c as i32 <= '9' as i32 {
-        return c as i32 - '0' as i32;
+
+pub fn xtoi(c: u8) -> i32 {
+    if c >= b'0' && c <= b'9' {
+        return (c - b'0') as i32;
     }
-    if c as i32 >= 'A' as i32 && c as i32 <= 'F' as i32 {
-        return c as i32 - 'A' as i32 + 10i32;
+    if c >= b'A' && c <= b'F' {
+        return (c - b'A' + 10) as i32;
     }
-    if c as i32 >= 'a' as i32 && c as i32 <= 'f' as i32 {
-        return c as i32 - 'a' as i32 + 10i32;
+    if c >= b'a' && c <= b'f' {
+        return (c - b'a' + 10) as i32;
     }
-    -1i32
+    -1
 }
```

Comment:
Maybe it would be better to use `i32::from_str_radix` of standard library.

Reply:
These function are not equivalent